common/src/main/java/com/selfdot/cobblemonmegas/common/command/CommandTree.java

```diff
@@ -23,96 +23,100 @@
-                    argument("players", EntityArgumentType.players())
-                    .executes(new GiveMegaStoneCommand())
+                        literal("givemegastone")
+                .requires(source ->
+                        !CobblemonMegas.getInstance().isDisabled() &&
+                                CobblemonMegas.getInstance().getPermissionValidator().hasPermission(source, GIVE_MEGA_STONE)
+                )
+                .then(RequiredArgumentBuilder.<ServerCommandSource, String>
+                                argument("megaStone", string())
+                        .suggests(CommandTree::megaStoneIDSuggestions)
+                        .then(RequiredArgumentBuilder.<ServerCommandSource, EntitySelector>
+                                        argument("players", EntityArgumentType.players())
+                                .executes(new GiveMegaStoneCommand())
+                        )
                 )
-            )
         );
         dispatcher.register(LiteralArgumentBuilder.<ServerCommandSource>
-            literal("givekeystone")
-            .requires(source ->
-                !CobblemonMegas.getInstance().isDisabled() &&
-                CobblemonMegas.getInstance().getPermissionValidator().hasPermission(source, GIVE_KEY_STONE)
-            )
-            .then(RequiredArgumentBuilder.<ServerCommandSource, String>
-                argument("keyStone", string())
-                .suggests((context, builder) -> {
-                    GiveKeyStoneCommand.KEY_STONE_TYPES.keySet().forEach(builder::suggest);
-                    return builder.buildFuture();
-                })
-                .then(RequiredArgumentBuilder.<ServerCommandSource, EntitySelector>
-                    argument("players", EntityArgumentType.players())
-                    .executes(new GiveKeyStoneCommand())
+                        literal("givekeystone")
+                .requires(source ->
+                        !CobblemonMegas.getInstance().isDisabled() &&
+                                CobblemonMegas.getInstance().getPermissionValidator().hasPermission(source, GIVE_KEY_STONE)
+                )
+                .then(RequiredArgumentBuilder.<ServerCommandSource, String>
+                                argument("keyStone", string())
+                        .suggests((context, builder) -> {
+                            GiveKeyStoneCommand.KEY_STONE_TYPES.keySet().forEach(builder::suggest);
+                            return builder.buildFuture();
+                        })
+                        .then(RequiredArgumentBuilder.<ServerCommandSource, EntitySelector>
+                                        argument("players", EntityArgumentType.players())
+                                .executes(new GiveKeyStoneCommand())
+                        )
                 )
-            )
         );
+        dispatcher.register(LiteralArgumentBuilder.<ServerCommandSource>
```

Comment:
Did not include permission node for this command as all users are expected to be able to use it, you may want to include one yourself

Reply:
I think that's fine.